src/d3a/d3a_core/sim_results/export_unmatched_loads.py

```diff
@@ -15,127 +15,171 @@
-    # Calculate overall metrics for the whole grid
-    unmatched_loads_result["unmatched_load_count"] = \
-        sum([v["unmatched_load_count"] for k, v in area_tree.items()])
-    unmatched_loads_result["all_loads_met"] = (unmatched_loads_result["unmatched_load_count"] == 0)
-    area_tree = OrderedDict(sorted(area_tree.items()))
-    unmatched_loads_result["areas"] = area_tree
-    return unmatched_loads_result
+        self.hour_list = hour_list()
+        # This is for only returning data until the current time_slot:
+        if hasattr(area, "past_markets") and len(list(area.past_markets)) > 0:
+            self.latest_time_slot = list(area.past_markets)[-1].time_slot
+        else:
+            self.latest_time_slot = self.hour_list[0]
+        self.name_uuid_map = {area.name: area.uuid}
+        self.area = area
+
+    def __call__(self):
+
+        unmatched_loads = self.arrange_output(
+            self.expand_to_ul_to_hours(
+                self.expand_ul_to_parents(
+                    self.find_unmatched_loads(self.area, {})[self.area.name],
+                    self.area.name, {})),
+            self.area)
+
+        return unmatched_loads, self.change_name_to_uuid(unmatched_loads)
+
+    def find_unmatched_loads(self, area, indict):
+        """
+        Aggregates list of times for each unmatched time slot for each load
+        """
+        indict[area.name] = {}
+        for child in area.children:
+            self.name_uuid_map[child.name] = child.uuid
+            if child.children:
+                indict[area.name] = self.find_unmatched_loads(child, indict[area.name])
+            else:
+                if isinstance(child.strategy, LoadHoursStrategy):
+                    indict[area.name][child.name] = \
+                        self._calculate_unmatched_loads_leaf_area(child)
+        return indict
+
+    @classmethod
+    def _calculate_unmatched_loads_leaf_area(cls, area):
+        """
+        actually determines the unmatched loads
+        """
+        unmatched_times = []
+        for market in area.parent.past_markets:
+            desired_energy_Wh = area.strategy.state.desired_energy_Wh[market.time_slot]
+            traded_energy_kWh = market.traded_energy[area.name] \
+                if market is not None and (area.name in market.traded_energy) \
+                else 0.0
+            deficit = desired_energy_Wh + traded_energy_kWh * 1000.0
+            if deficit > FLOATING_POINT_TOLERANCE:
+                unmatched_times.append(market.time_slot)
+        return {"unmatched_times": unmatched_times}
+
+    def change_name_to_uuid(self, indict):
+        """
+        postprocessing: changing area names to uuids
+        """
+        new = {}
+        for k, v in indict.items():
+            new[self.name_uuid_map[k]] = v
+        return new
+
+    def _accumulate_all_uls_in_branch(self, subdict, unmatched_list) -> list:
+        """
+        aggregate all UL times in one branch into one list
```

Comment:
I guess its okay concatenate the unmatched_loads to uls in method naming. But in comments it would be better to use full name for better understanding